Base code with futures (fixes #3)

mkgerasimenko opened the pull request:
- added gradle quality plugin;
- added TestNG listener;
- added test-data-supplier library.

mkgerasimenko committed:
Added and changed the following:
 - added gradle quality plugin;
 - fixed all violations;
 - added TestNG listener;
 - added test-data-supplier library;
 - created new tests;
 - changed gradle version up to the newer.

sskorol commented on src/main/java/com/waverleysoftware/model/Panel.java:
```diff
@@ -0,0 +1,43 @@
+package com.waverleysoftware.model;
+
+/**
+ * A simple model class for user processing.
```
It doesn't seem to be a class for user processing. ;)

sskorol commented on src/main/java/com/waverleysoftware/model/Panel.java:
```diff
@@ -0,0 +1,43 @@
+        return type;
+    }
+
+    public boolean isHasTouchScreen() {
```
`isHas` sounds weird. Use the better.

sskorol commented on src/main/java/com/waverleysoftware/model/Sensor.java:
```diff
@@ -0,0 +1,23 @@
+package com.waverleysoftware.model;
+
+/**
+ * Author mkgerasimenko.
```
Add a more straightforward javadoc. Don't leave a default one.

sskorol commented on src/main/java/com/waverleysoftware/model/Type.java:
```diff
@@ -0,0 +1,9 @@
+package com.waverleysoftware.model;
+
+/**
+ * Author mkgerasimenko.
```
Leaving default javadoc is not a good practice.

sskorol commented on src/test/java/com/waverleysoftware/AccessCodeTests.java:
```diff
@@ -1,32 +1,34 @@
+import java.util.logging.Level;
+
+/**
+ * Author mkgerasimenko.
```
See above comments regarding javadocs.

sskorol commented on src/test/java/com/waverleysoftware/BaseTest.java:
```diff
@@ -3,14 +3,22 @@
+import java.util.logging.Logger;
+
+/**
+ * Author mkgerasimenko.
```
See comments about javadocs.

sskorol reviewed (changes requested):
@mkgerasimenko check comments please

mkgerasimenko committed:
Added and changed the following:
 - added sample javadoc for both main and test packages;
 - added one more TestNG listener;
 - common refactoring.

sskorol commented on src/main/java/com/waverleysoftware/listemers/SecondTestListener.java:
```diff
@@ -0,0 +1,22 @@
+@SuppressWarnings("JavadocType")
+public class SecondTestListener implements ISuiteListener {
+
+    private static final Logger LOG = Logger.getLogger(TestListener.class.getName());
```
try replacing common java logger with `@Slf4j` annotation, which is a part of Lombok library

mkgerasimenko replied:
I did it in the  **5-lombok-assert-annotation** branch. Please, look at the changes.

sskorol commented on src/main/java/com/waverleysoftware/model/Panel.java:
```diff
@@ -0,0 +1,43 @@
+/**
+ * A simple model class for panel processing.
+ */
+public class Panel {
```
use Lombok to remove boilerplate for all entities

mkgerasimenko replied:
I did it in the  **5-lombok-assert-annotation** branch. Please, look at the changes.

sskorol commented on src/test/java/com/waverleysoftware/AccessCodeTests.java:
```diff
@@ -1,32 +1,31 @@
+        final String conditionForType = "home";
+        if (conditionForType.equals(panel.getType())
+                && panel.hasTouchScreen()
+                && LOG.isLoggable(Level.INFO)) {
```
with `@Slf4j` you don't need to check of corresponding logging type is available

mkgerasimenko replied:
I did it in the  **5-lombok-assert-annotation** branch. Please, look at the changes.

sskorol reviewed (changes requested):
@mkgerasimenko  check comments please

sskorol commented:
LGTM
